sdk/nanotdf.go

```diff
@@ -749,6 +749,11 @@ func (s SDK) CreateNanoTDF(writer io.Writer, reader io.Reader, config NanoTDFCon
 
 // ReadNanoTDF - read the nano tdf and return the decrypted data from it
 func (s SDK) ReadNanoTDF(writer io.Writer, reader io.ReadSeeker) (uint32, error) {
+	return s.ReadNanoTDFContext(context.Background(), writer, reader)
+}
+
+// ReadNanoTDFContext - allows cancelling the reader
+func (s SDK) ReadNanoTDFContext(ctx context.Context, writer io.Writer, reader io.ReadSeeker) (uint32, error) {
```

Comment:
@dmihalcik-virtru  Are we planning on deprecating `ReadNanoTDF` or should we just break the method signature and add the context as the first argument.

Reply:
I was modeling this after the `slog.[Level]Context` [methods](https://pkg.go.dev/golang.org/x/exp/slog#hdr-Contexts)

Reply:
So no I was not planning to deprecate. We should encourage use of the Context variant wherever possible (e.g. within a context-having thread of execution)